sql/core/src/main/scala/org/apache/spark/sql/execution/command/commands.scala

```diff
@@ -423,6 +424,100 @@ case class ShowTablePropertiesCommand(
 }
 
 /**
+ * A command for users to list the column names for a table. This function creates a
+ * [[ShowColumnsCommand]] logical plan.
+ *
+ * The syntax of using this command in SQL is:
+ * {{{
+ *   SHOW COLUMNS (FROM | IN) table_identifier [(FROM | IN) database];
+ * }}}
+ */
+case class ShowColumnsCommand(table: TableIdentifier) extends RunnableCommand {
+  // The result of SHOW COLUMNS has one column called 'result'
+  override val output: Seq[Attribute] = {
+    AttributeReference("result", StringType, nullable = false)() :: Nil
+  }
+
+  override def run(sqlContext: SQLContext): Seq[Row] = {
+    val relation = sqlContext.sessionState.catalog.lookupRelation(table, None)
+    relation.schema.fields.map { field =>
+      Row(field.name)
+    }
```

Comment:
@andrewor14 Hi Andrew, actually i remember i had tried this before. The table metadata's schema does not include partition columns ? Thats why i had used the lookupRelation.. 

Please let me know what you think ..